nntrainer/layers/layer_node.cpp

```diff
@@ -433,6 +470,18 @@ float LayerNode::getLoss() const {
   return *loss;
 }
 
+void LayerNode::setInputDimension(const TensorDim &dim, unsigned int idx) {
+  NNTR_THROW_IF(idx >= getNumInputs(), std::out_of_range)
+    << "Setting dimensions out of bounds, idx: " << idx
```

Comment:
Just for the double check, getNumInputs is equal to the number of connections for now.

Reply:
Yes, this is true :)